README.md

```diff
@@ -131,40 +150,149 @@ FAROS_VCS_ORG="<vcs_org>" \
+    The unique identifier of the build that constructed the artifact being deployed.
+
+> Deployment Optional Fields
+
+4. `FAROS_DEPLOYMENT` / `--deployment "<deployment>"`  
+    The unique id of the deployment.  
+    __Default__: Random UUID  
+  
+1. `FAROS_DEPLOYMENT_ENV_DETAILS` / `--deployment_env_details "<details>"`  
+    Any additional details about the deployment environment that you wish to provide.  
+    __Default__: ""  
+  
+1. `FAROS_DEPLOYMENT_STATUS_DETAILS` / `--deployment_status_details "<details>"`  
+    Any additional details about the status of the deployment that you wish to provide.  
+    __Default__: ""  
+  
+1. `FAROS_SOURCE` / `--source "<source>"`  
+    The source that will be associate with the deployment  
+    __Default__: "Faros_Script"  
 
-##### Using flags
+##### :mega: Sending a deployment event examples
+
+Using flags
 
 ```sh
 ./faros_event.sh deployment -k "<api_key>" \
     --app "<app_name>" \
     --ci_org "<ci_organization>" \
     --ci_source "<ci_source>" \
-    --commit_sha "<commit_sha>" \
     --deployment_status "<deploy_status>" \
     --deployment_env "<environment>" \
     --pipeline "<ci_pipeline>" \
     --build "<build>"
 ```
 
-##### Or using environment variables
+Or using environment variables
 
 ```sh
 FAROS_API_KEY="<api_key>" \
 FAROS_APP="<app_name>" \
 FAROS_CI_ORG="<ci_org>" \
 FAROS_CI_SOURCE="<ci_source>" \
-FAROS_COMMIT_SHA="<commit_sha>" \
 FAROS_DEPLOYMENT_STATUS="<deploy_status>" \
 FAROS_DEPLOYMENT_ENV="<environment>" \
 FAROS_PIPELINE="<pipeline>" \
 FAROS_BUILD="<build>" \
 ./faros_event.sh deployment
 ```
 
-#### :mega: Sending a build_deployment (build and deployment) event
-
-##### Using flags
+---
+
+#### Build and Deployment Event - `build_deployment`
+
+The `build_deployment` should be used when there is not a distinct build that created the artifact that is being deployed. In order for Faros to associate the code that is being deployed, a build that links a commit sha to the deployment will be created.
+
+> Build and Deployment Required Fields
+
+1. `FAROS_DEPLOYMENT_ENV` / `--deployment_env "<environment>"`  
+    The environment that the application is being deployed to.  
+    __Allowed Values:__ Prod, Staging, QA, Dev, Sandbox, Custom
+  
+1. `FAROS_DEPLOYMENT_STATUS` / `--deployment_status "<status>"`  
+    The status of the deployment.  
+    __Allowed Values:__ Success, Failed, Canceled, Queued, Running, RolledBack, Custom
+  
+1. `FAROS_BUILD_STATUS` / `--build_status "<build_status>"`  
+    The status of the build.  
+    __Allowed Values:__ Success, Failed, Canceled, Queued, Running, Unknown, Custom
+  
+1. `FAROS_VCS_SOURCE` / `--vcs_source "<vcs_source>"`  
+    The version control source system that stores the code that is being built (i.e. GitHub, GitLab, Bitbucket) Please note that this field is case sensitive. If you have a feed that connects to one of these sources, this name must match exactly to be correctly associated.
```

Comment:
I created #14 to actually flesh out these source systems etc to mitigate issues from capitalization etc. The PR is already kinda big tho so I was just going to do it in a subsequent PR.